core/src/main/scala/cats/data/IndexedReaderWriterStateT.scala

```diff
@@ -831,11 +892,15 @@ private trait RWSTAlternative1[F[_], E, L, S]
 
   def empty[A]: ReaderWriterStateT[F, E, L, S, A] = ReaderWriterStateT.liftF(G.empty[A])
 
-  def pure[A](a: A): ReaderWriterStateT[F, E, L, S, A] = ReaderWriterStateT.pure[F, E, L, S, A](a)
+  // Must exist in this trait for binary compatibility.
+  override def pure[A](a: A): ReaderWriterStateT[F, E, L, S, A] =
+    pure0(a)
```

Comment:
Instead of `pure0` can you just call `super.pure` or something?

Reply:
Yeah, It was the first thing I tried, actually. But turned out it mysteriously broke binary compatibility 🤷 
I don't remember all the details already, but if you're concerned about that, I can give it one more shot and see what happens. And then I can let you know why in more details :)

Reply:
If you don't mind I'd be curious, I'm a bincompat nerd 🤓

Reply:
Ok, I added a commit that rolls back the hack for the binary compatibility.
Let's see how it goes now...

Reply:
..and it begins failing as anticipated :)
Looks like if we simply call a method from super, then the compiler simply optimizes the callee out somehow.

Reply:
So these are the errors? Actually they are safe to exclude, assuming that `RWSTAlternative1` cannot be extended outside of Cats which I believe is the case.

```
[error]  * abstract synthetic method cats$data$RWSTAlternative1$$super$pure(java.lang.Object)cats.data.IndexedReaderWriterStateT in interface cats.data.RWSTAlternative1 is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("cats.data.RWSTAlternative1.cats$data$RWSTAlternative1$$super$pure")
[error]  * abstract synthetic method cats$data$RWSTAlternative1$$super$ap(cats.data.IndexedReaderWriterStateT,cats.data.IndexedReaderWriterStateT)cats.data.IndexedReaderWriterStateT in interface cats.data.RWSTAlternative1 is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("cats.data.RWSTAlternative1.cats$data$RWSTAlternative1$$super$ap")
```

Reply:
Ah, that's good then, will do shortly. Thank you for looking into it and helping me with the concern!

Reply:
fixed